src/wp-includes/class-wp-theme-json.php

```diff
@@ -2886,4 +2907,113 @@ public function get_data() {
 		return $output;
 	}
 
+	/**
+	 * Sets the spacingSizes array based on the spacingScale values from theme.json.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @return void
+	 */
+	public function set_spacing_sizes() {
+		$spacing_scale = _wp_array_get( $this->theme_json, array( 'settings', 'spacing', 'spacingScale' ), array() );
+
+		if ( ! is_numeric( $spacing_scale['steps'] )
+			|| ! isset( $spacing_scale['mediumStep'] )
+			|| ! isset( $spacing_scale['unit'] )
+			|| ! isset( $spacing_scale['operator'] )
+			|| ! isset( $spacing_scale['increment'] )
+			|| ! isset( $spacing_scale['steps'] )
+			|| ! is_numeric( $spacing_scale['increment'] )
+			|| ! is_numeric( $spacing_scale['mediumStep'] )
+			|| ( '+' !== $spacing_scale['operator'] && '*' !== $spacing_scale['operator'] ) ) {
+			if ( ! empty( $spacing_scale ) ) {
+				trigger_error( __( 'Some of the theme.json settings.spacing.spacingScale values are invalid' ), E_USER_NOTICE );
+			}
+			return null;
+		}
+
+		// If theme authors want to prevent the generation of the core spacing scale they can set their theme.json spacingScale.steps to 0.
+		if ( 0 === $spacing_scale['steps'] ) {
+			return null;
+		}
+
+		$unit            = '%' === $spacing_scale['unit'] ? '%' : sanitize_title( $spacing_scale['unit'] );
+		$current_step    = $spacing_scale['mediumStep'];
+		$steps_mid_point = round( $spacing_scale['steps'] / 2, 0 );
+		$x_small_count   = null;
```

Comment:
Since `$x` has been renamed to `$below_midpoint_count`, `$x_small_count` should be renamed to make it clear what it refers to. Does `$small_below_midpoint_count` make sense here, or?

Same for `$x_large_count`.

Reply:
in this case the `x` relates to the number of `X`s in the X-Small count, eg. `X-Small, 2X-Small` so it sort of is meaningful rather than just a standard `$x` count variable .... still waiting on a decision if we retain this or not, but can rename it if we do retain it and you think that is still confusing

Reply:
We are  retaining the t-shirt style sizing for spacing scales with steps > 7 for now, but with the `$x` vars gone now, and the var used in this context `sprintf( __( '%sX-Small' ), (string) $x_small_count ),` I think the naming is clear enough to avoid confusion around it ... but let me know if you still think it should be renamed.